LongModuleFinder/main.cpp

```diff
@@ -0,0 +1,129 @@
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <CppBuildInsights.hpp>
+
+using namespace Microsoft::Cpp::BuildInsights;
+using namespace Activities;
+using namespace SimpleEvents;
+
+class LongModuleFinder : public IAnalyzer
+{
+    struct IdentifiedCompilerInvocation
+    {
+        unsigned InvocationId;
+        double InvocationTime;
+
+        bool operator<(const IdentifiedCompilerInvocation& other) const {
+            return InvocationTime > other.InvocationTime;
+        }
+    };
+
+public:
+    LongModuleFinder() :
+        pass_{ 0 },
+        cachedInvocationDurations_{},
+        identifiedCompilerInvocations_{}
+    {}
+
+    AnalysisControl OnBeginAnalysisPass() override
+    {
+        ++pass_;
+        return AnalysisControl::CONTINUE;
+    }
+
+    AnalysisControl OnStopActivity(const EventStack& eventStack)
+        override
+    {
+        if (pass_ == 1) {
+            MatchEventStackInMemberFunction(eventStack, this,
+                &LongModuleFinder::OnStopCompiler);
+        }
+
+        return AnalysisControl::CONTINUE;
+    }
+
+    AnalysisControl OnSimpleEvent(const EventStack& eventStack) override
+    {
+        if (pass_ == 2) {
+            MatchEventStackInMemberFunction(eventStack, this,
+                &LongModuleFinder::OnModuleEvent);
+        }
+
+        return AnalysisControl::CONTINUE;
+    }
+
+    void OnStopCompiler(Compiler cl)

```

Comment:
`Compiler` represents an entire compiler invocation which I think can include multiple modules, header units, etc. To get the individual time for each module or header unit, `FrontEndPass` would be more reliable. By using the `FrontEndPass`, you also have access to the module, header unit, or PCHs file name which is more useful than the compiler invocation when writing out the longest items.

Reply:
Fixed.